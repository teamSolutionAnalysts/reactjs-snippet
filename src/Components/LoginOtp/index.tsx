import React from "react";
import { Formik, Form } from "formik";
import { withTranslation } from "@Server/i18n";
import { NextPage } from "next";
import { STLInput } from "@Components/Common";
import OTPInput from "@Components/Common/OtpInput";
import style from "../Login/Login.module.scss";

interface Props {
    title?: string;
    subtitle?: string;
    username?: any;
    verifyOTP?: any;
    isVarified?: boolean;
    onNext?: any;
    onResend?: any;
    verifyOTPSuccess?: any;
    verifyOTPError?: any;
    setOtpVerifyMsg?: any;
    otpLengthConfig?: string;
    otpValidTime?: string;
    resendMaxAttemp?: string;
    resendMaxAttempCount?: any;
}

const LoginOtp: NextPage<Props> = ({
    t,
    title,
    subtitle,
    username,
    verifyOTP,
    isVarified,
    onNext,
    onResend,
    verifyOTPSuccess,
    verifyOTPError,
    setOtpVerifyMsg,
    otpLengthConfig,
    otpValidTime,
    resendMaxAttemp,
    resendMaxAttempCount,
}: any) => {
    const [otpFieldData, setOTPData] = React.useState<any>();
    const [otpError, setOTPError] = React.useState(false);
    const [isResend, setIsResend] = React.useState(false);

    // function to change the input of the otp on change
    const changeOtp = (otpData: any) => {
        if (!otpData) {
            setOtpVerifyMsg("");
        } else {
            setOTPError(false);
        }
        setOTPData(otpData);
    };

    return (
        <>
            <h2 className={`${style.h2}`}>{t(title)}</h2>
            <h3 className={`${style.h3}`}>{t(subtitle)}</h3>
            {/* formik form for the username and OTP value */}
            <Formik
                initialValues={{
                    username: username || "",
                }}
                onSubmit={() => {
                    if (otpFieldData) {
                        verifyOTP(otpFieldData, username);
                    } else {
                        setOTPError(true);
                    }
                }}
            >
                {({ values }) => (
                    <Form className={`${style["login-from"]}`}>
                        <div className="form-group">
                            <label>{t("loginotp:Username")}*</label>
                            {/* customised input for username */}
                            <STLInput
                                className={`${style["form-control"]}`}
                                placeholder={t("loginotp:Username")}
                                type="text"
                                name="username"
                                value={values.username}
                            />
                        </div>
                        <div className={`${style["otp-section"]}`}>
                            {/* customised input for OTP */}
                            <OTPInput
                                autoFocus
                                isNumberInput
                                length={Number(otpLengthConfig)}
                                className={style.optContainer}
                                inputClassName={style.otpInput}
                                onChangeOTP={(otpData: any) => changeOtp(otpData)}
                                isResend={isResend}
                                setIsResend={setIsResend}
                            />
                            {/* Error messages */}
                            {otpError && (
                                <div className={`${style["invalid-username-txt"]}`}>
                                    {t("formikError:OTP_REQUIRED")}
                                </div>
                            )}
                            {isVarified && verifyOTPSuccess && (
                                <div className={`${style["valid-username-txt"]}`}>{verifyOTPSuccess}</div>
                            )}
                            {!isVarified && verifyOTPError && (
                                <div className={`${style["invalid-username-txt"]}`}>{verifyOTPError}</div>
                            )}
                        </div>
                        <div className={`${style.expiretxt} mt-2`}>
                            {t(`loginotp:YOUR_OTP_WILL_EXPIRE_IN_60_SECONDS`, { otpValidTime })}
                        </div>
                        <div className={style["resend-otp-section"]}>
                            <p>
                                {t("loginotp:NotReceivedYourCode?")}
                                <span
                                    className={
                                        resendMaxAttempCount > resendMaxAttemp
                                            ? `${style.disabled} ml-1 mr-1`
                                            : "ml-1 mr-1"
                                    }
                                >
                                    <a
                                        onClick={() => {
                                            setIsResend(true);
                                            onResend(values.username);
                                        }}
                                    >
                                        {t("loginotp:ResendCode")}
                                    </a>
                                </span>
                            </p>
                        </div>

                        <div className="text-center mb-4">
                            {!isVarified ? (
                                <button type="submit" className={`${style.btn} ${style["btn-large"]}`}>
                                    Verify{" "}
                                </button>
                            ) : (
                                <button
                                    type="button"
                                    className={`${style.btn} ${style["btn-large"]}`}
                                    onClick={() => onNext()}
                                >
                                    Next{" "}
                                </button>
                            )}
                        </div>
                    </Form>
                )}
            </Formik>
        </>
    );
};

export default withTranslation(["login", "formikError", "loginotp"])(LoginOtp);
