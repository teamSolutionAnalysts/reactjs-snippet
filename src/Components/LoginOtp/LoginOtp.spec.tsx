// #region Global Imports
import * as React from "react";
// #endregion Global Imports

// #region Local Imports
import { render } from "@Test/utils";
import LoginOtp from ".";
// #endregion Local Imports

describe("LoginOtp", () => {
    it("should match snapshot", () => {
        const { container } = render(<LoginOtp />);
        expect(container).toMatchSnapshot();
    });
});
