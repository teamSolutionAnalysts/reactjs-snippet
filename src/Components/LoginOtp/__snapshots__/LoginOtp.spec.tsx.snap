// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginOtp should match snapshot 1`] = `
<div>
  <h2
    class="undefined"
  />
  <h3
    class="undefined"
  />
  <form
    action="#"
    class="undefined"
  >
    <div
      class="form-group"
    >
      <label>
        loginotp:Username
        *
      </label>
      <div>
        <input
          class="ant-input undefined undefined undefined"
          name="username"
          placeholder="loginotp:Username"
          type="text"
          value=""
        />
        <span
          class="ant-typography ant-typography-danger undefined undefined"
          data-testid="form-error"
        />
      </div>
    </div>
    <div
      class="undefined"
    >
      <div>
        <input
          value=""
        />
        <input
          value=""
        />
        <input
          value=""
        />
        <input
          value=""
        />
        <input
          value=""
        />
        <input
          value=""
        />
      </div>
    </div>
    <div
      class="undefined"
    >
      <p>
        loginotp:NotReceivedYourCode?
        <a
          class="ml-1 mr-1"
        >
          loginotp:ResendCode
        </a>
      </p>
    </div>
    <div
      class="text-center mb-4"
    >
      <button
        class="undefined undefined"
        type="submit"
      >
        Verify
         
      </button>
    </div>
  </form>
</div>
`;
